### PR TITLE
Avoid false positive in EntryPointInfo and TypePropertyCacheElement

### PR DESCRIPTION
**Pull request by curtisman**
Add a tag to some integral data in EntryPointInfo and TypePropertyCacheElement to avoid false positive GC pointer.

### CODE REVIEW

**Comment by curtisman** [comment]
@digitalinfinity please review.

File: lib/Runtime/Base/FunctionBody.h [hunk 1]
```diff
@@ -650,7 +645,7 @@ namespace Js
             jitTransferData(nullptr), sharedPropertyGuards(nullptr), propertyGuardCount(0), propertyGuardWeakRefs(nullptr),
             equivalentTypeCacheCount(0), equivalentTypeCaches(nullptr), constructorCaches(nullptr), state(NotScheduled), inProcJITNaticeCodedata(nullptr),
             numberChunks(nullptr), numberPageSegments(nullptr), polymorphicInlineCacheInfo(nullptr), runtimeTypeRefs(nullptr),
-            isLoopBody(isLoopBody), hasJittedStackClosure(false), registeredEquivalentTypeCacheRef(nullptr), bailoutRecordMap(nullptr),
+            tag(1), isLoopBody(isLoopBody), hasJittedStackClosure(false), registeredEquivalentTypeCacheRef(nullptr), bailoutRecordMap(nullptr),
```
Comment (obastemur) [inline lib/Runtime/Base/FunctionBody.h]: `tag` here needs to be defined for both JIT and no-jit builds.

Comment (curtisman) [inline lib/Runtime/Base/FunctionBody.h]: Fixed.

**Comment by digitalinfinity** [comment]
:shipit:

**Pull request merged**